src/App.jsx

```diff
@@ -14,45 +16,100 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
-export const App = () => (
-  <div className="section content">
-    <div className="buttons">
-      <button
-        type="button"
-        className="button is-info is-light"
-      >
-        Sort alphabetically
-      </button>
-
-      <button
-        type="button"
-        className="button is-success is-light"
-      >
-        Sort by length
-      </button>
-
-      <button
-        type="button"
-        className="button is-warning is-light"
-      >
-        Reverse
-      </button>
-
-      <button
-        type="button"
-        className="button is-danger is-light"
-      >
-        Reset
-      </button>
-    </div>
+const SORT_FIELD_LENGTH = 'length';
+const SORT_FIELD_ALPHA = 'alphabetically';
+
+let goodsCopy = [...goodsFromServer];
+
+function getPreparedGoods(goods, { sortField, reverse }) {
+  let preparedGoods = [...goods];
+
+  switch (sortField) {
+    case SORT_FIELD_LENGTH:
+      preparedGoods = preparedGoods
+        .sort((elem1, elem2) => elem1.length - elem2.length);
+      break;
+    case SORT_FIELD_ALPHA:
+      preparedGoods = preparedGoods
+        .sort((elem1, elem2) => elem1.localeCompare(elem2));
+      break;
+
+    default:
+      preparedGoods = goodsFromServer;
+  }
+
+  if (reverse && sortField) {
+    preparedGoods = preparedGoods.reverse();
+  }
+
+  return preparedGoods;
+}
+
+export const App = () => {
+  const [sortField, setSortFeild] = useState('');
+  const [reverse, setReverse] = useState(false);
 
-    <ul>
-      <li data-cy="Good">Dumplings</li>
-      <li data-cy="Good">Carrot</li>
-      <li data-cy="Good">Eggs</li>
-      <li data-cy="Good">Ice cream</li>
-      <li data-cy="Good">Apple</li>
-      <li data-cy="Good">...</li>
-    </ul>
-  </div>
-);
+  goodsCopy = getPreparedGoods(goodsCopy, { sortField, reverse });
+
+  return (
+    <div className="section content">
+      <div className="buttons">
+        <button
+          type="button"
+          className={
+            cn('button is-info', { 'is-light': sortField !== SORT_FIELD_ALPHA })
+          }
+          onClick={() => setSortFeild(SORT_FIELD_ALPHA)}
+        >
+          Sort alphabetically
+        </button>
+
+        <button
+          type="button"
+          onClick={() => setSortFeild(SORT_FIELD_LENGTH)}
+          className={
+            cn(
+              'button is-success',
+              { 'is-light': sortField !== SORT_FIELD_LENGTH },
+            )
+          }
+        >
+          Sort by length
+        </button>
+
+        <button
+          type="button"
+          onClick={() => {
+            if (sortField === '') {
+              goodsCopy = goodsCopy.reverse();
+            }
```

Comment:
why do you need it here, if you already have reverse logic in the getPreparedGoods